GNUmakefile

```diff
@@ -4,8 +4,10 @@ PKG_NAME=azurerm
 
 #make sure we catch schema errors during testing
 TF_SCHEMA_PANIC_ON_ERROR=1
-GO111MODULE=on
-GOFLAGS=-mod=vendor
+
+# Turn on Go modules and use local vendor copy by default
+export GO111MODULE=on
+export GOFLAGS=-mod=vendor
```

Comment:
Go Modules should be enabled within Travis since we get a `go get` error if we don't run the tooling command with `GO111MODULE=off` - as such I don't believe this change is necessary? is this a Windows specific bug?

Reply:
@metacpp your PR's running `go mod vendor` outside of the makefile [where this functionality is turned on](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/GNUmakefile#L6-L8) - as such this functionality won't be enabled outside of the Makefile.

Reply:
Yes, the above Travis result link is to `prove` `GO111MODULE` is disabled by default.

I did another change to test running of `go mod` in GNUMakefile. And it is also not working:
https://travis-ci.org/terraform-providers/terraform-provider-azurerm/builds/483537544?utm_source=github_status&utm_medium=notification (edited) 

Please take a look at this link: 
https://stackoverflow.com/questions/27282660/setting-environment-variables-in-a-makefile

Each line of recipe will be a sub-shell instance, to make things simpler here, we'd better just `export` them in GNUMakefile.